### PR TITLE
DataConvertible protocol, remove NSData references

### PR DESCRIPTION
**Pull request by paulrolfe**
DataConvertible is an easy way to get from number types to Data and back. It could apply to UnsignedIntegers as well, but we already had the capability built in to them without the need for an optional initializer, so UnsignedIntegers actually don't technically conform to the protocol, although they have something very similar.

And I'd also like to remove all the `NSData` conveniences we built. I think everyone should have migrated to using `Data` by now. If this is too extreme to remove so much, I can hold off on that.

Will bump version to `0.7.0`.

### CODE REVIEW

File: SwiftWisdom/Core/Foundation/Data/DataConvertible.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
+//
+//  DataConvertible.swift
+//  SwiftWisdom
+//
+//  Created by Paul Rolfe on 2/3/17.
+//  Copyright © 2017 Intrepid. All rights reserved.
+//
+
```
Comment (paulrolfe) [inline SwiftWisdom/Core/Foundation/Data/DataConvertible.swift]: I'll add a note that I borrowed this protocol from http://stackoverflow.com/questions/38023838/round-trip-swift-number-types-to-from-data

**Review by alexpersian — APPROVED** [review]
Looks good. Might be useful to get a 2nd opinion on the `NSData` removal.

**Review by alexpersian — CHANGES_REQUESTED** [review]
Fix failing tests.

**Comment by paulrolfe** [comment]
retest this please

**Review by alexpersian — APPROVED** [review]

**Comment by bobgilmore** [comment]
Are you cool not having a testpoint to exercise `ip_int8Value`? It's a pretty thin wrapper around `.to()` so I wouldn't be *crushed* if it's not exercised, but figured I'd point it out.

**Comment by brightredchilli** [comment]
let's see how this one goes, but in the future, I think we should experiment with deprecating some versions before removing.

**Review by brightredchilli — APPROVED** [review]

**Comment by paulrolfe** [comment]
@bobgilmore  yeah might as well test it. Just added that.

**Pull request merged**